app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -13,15 +13,8 @@ def download_file
       filename = ForemanInventoryUpload.facts_archive_name(params[:organization_id])
       files = Dir["{#{ForemanInventoryUpload.uploads_file_path(filename)},#{ForemanInventoryUpload.done_file_path(filename)}}"]
 
-      return send_file files.first, disposition: 'attachment', filename: filename unless files.empty?
-      throw_flash_error "File doesn't exist"
-    end
-
-    def throw_flash_error(message)
-      process_error(
-        :redirect => foreman_inventory_upload_index_path,
-        :error_msg => message
-      )
+      return render json: { message: "File doesn't exist" }, status: :not_found if files.empty?
+      render :json => File.read(files.first)
```

Comment:
I have a better solution: Let's leave the browser to handle the download in a separate page.
If there will be an error, instead of rendering a flash error, we will render a 404, or a specific "file not found" page.

Reply:
After all of the hacks? :angel: just kidding,
I like this approach too,
and I decided to go with `raise ActionController::BadRequest.new("The report file doesn't exist")`:
![Screenshot (50)](https://user-images.githubusercontent.com/26363699/107554657-edb66f00-6bde-11eb-9d55-6cd53c28d5fa.png)

instead of just a plain empty page with a small message:
![Screenshot (51)](https://user-images.githubusercontent.com/26363699/107554677-f27b2300-6bde-11eb-827f-3f00d2ee45a6.png)